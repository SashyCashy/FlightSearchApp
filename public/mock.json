[
  {
    "name": "Carrie Grabowski",
    "id": 1,
    "age": "30",
    "address": ["1585 Live Oak Rd #98 Vista CA 92081"],
    "image": "female.png"
  },
  {
    "name": "James Cameroon",
    "id": 2,
    "age": "24",
    "address": ["Maya Linda Rd San Diego CA 92126"],
    "image": "male.png"
  },
  {
    "name": "Alaleh Baghini",
    "id": 3,
    "age": "39",
    "address": ["2848 Whiptail Road Carlsbad CA 92081"],
    "image": "female.png"
  },
  {
    "name": "Jimit Patel",
    "id": 4,
    "age": "35",
    "address": [
      "E-402, Reelicon Garden Grove Dattanagar Katraj Pune",
      "1585 Live Oak Rd #95 Vista CA 92081"
    ],
    "image": "male.png"
  },
  {
    "name": "Norif Takaya",
    "id": 5,
    "age": "42",
    "address": [],
    "image": "male.png"
  },
  {
    "name": "Carrie Grabowski",
    "id": 6,
    "age": "30",
    "address": ["1585 Live Oak Rd #98 Vista CA 92081"],
    "image": "female.png"
  },
  {
    "name": "James Cameroon",
    "id": 7,
    "age": "24",
    "address": ["Maya Linda Rd San Diego CA 92126"],
    "image": "male.png"
  },
  {
    "name": "Alaleh Baghini",
    "id": 8,
    "age": "39",
    "address": ["2848 Whiptail Road Carlsbad CA 92081"],
    "image": "female.png"
  },
  {
    "name": "Jimit Patel",
    "id": 9,
    "age": "35",
    "address": [
      "E-402, Reelicon Garden Grove Dattanagar Katraj Pune",
      "1585 Live Oak Rd #95 Vista CA 92081"
    ],
    "image": "male.png"
  },
  {
    "name": "Norif Takaya",
    "id": 10,
    "age": "42",
    "address": [],
    "image": "male.png"
  },
  {
    "name": "Carrie Grabowski",
    "id": 11,
    "age": "30",
    "address": ["1585 Live Oak Rd #98 Vista CA 92081"],
    "image": "female.png"
  }
]
