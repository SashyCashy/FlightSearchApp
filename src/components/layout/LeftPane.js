/**
 * File Description: This component is the left pane of the website to display the results of the customer
 * Author(s): Sashank Pindiproli
 * Date of Creation: 11/07/2020
 */

import React, { useState } from 'react';
import { ListGroup } from 'react-bootstrap';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const LeftPane = ({ customerList, makeDataRequest }) => {
  const [activeId, setActiveId] = useState(-1);

  /** FETCH THE LIST OF CUSTOMERS FROM customers.json */

  return (
    <StyledContainer id="left-pane__container">
      <ListGroup as="ul">
        {customerList.map(({ name, id, age }) => (
          <ListGroup.Item
            as="li"
            active={activeId === id}
            key={id}
            action
            onClick={() => {
              setActiveId(id);
              makeDataRequest(id);
            }}
          >
            <div>{id}</div>
            <div>{name}</div>
            <div>{age}</div>
          </ListGroup.Item>
        ))}
      </ListGroup>
    </StyledContainer>
  );
};

const StyledContainer = styled.div`
  width: calc(40%);
  display: flex;
  margin-top: 0.5vh;
  flex-direction: column;
  overflow: scroll;

  height: 91.5vh;
  .list-group-item {
    display: grid;
    grid-template-columns: 0.5fr 1fr 1fr;
    background: #78e3fd;
    font-size: 2em;
    font-weight: 500;
  }
`;

LeftPane.propTypes = {
  customerList: PropTypes.array.isRequired,
  makeDataRequest: PropTypes.func,
};

LeftPane.defaultProps = {
  customerList: [],
};

export default LeftPane;
