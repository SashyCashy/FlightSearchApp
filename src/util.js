const fetchCustomerList = async () => {
  let data = await fetch('customer.json');
  let result = await data.json();
  return result;
};

export { fetchCustomerList };
