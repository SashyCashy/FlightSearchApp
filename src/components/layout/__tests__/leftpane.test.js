import React from 'react';
import { mount, render } from 'enzyme';
import LeftPane from '../LeftPane';
import { fetchCustomerList } from '../../../util';
import mock from '../../../../../public/mock.json';
beforeEach(() => {
  fetch.resetMocks();
});

describe('Fetch data for left pane ', () => {
  it('Check the fetch call ', async () => {
    fetch.mockResponseOnce(JSON.stringify(mock));

    let customerList = await fetchCustomerList();
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toHaveBeenCalledWith(`customer.json`);
    expect(customerList).not.toBeNull();
    expect(customerList).toHaveLength(11);
  });

  test('Check if the values are being fetched or not ', async () => {
    fetch.mockResponseOnce(JSON.stringify(mock));

    let customerList = await fetchCustomerList();
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toHaveBeenCalledWith(`customer.json`);
    const mockCallback = jest.fn((x) => 42 + x);
    const wrapper = mount(
      <LeftPane
        customerList={customerList}
        makeDataRequest={() => mockCallback()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
