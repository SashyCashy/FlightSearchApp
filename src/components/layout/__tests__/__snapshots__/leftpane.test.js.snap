// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch data for left pane  Check if the values are being fetched or not  1`] = `
<div
  className="sc-AxjAm ibklLD"
  id="left-pane__container"
>
  <ul
    className="list-group"
    onKeyDown={[Function]}
  >
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        1
      </div>
      <div>
        Carrie Grabowski
      </div>
      <div>
        30
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        2
      </div>
      <div>
        James Cameroon
      </div>
      <div>
        24
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        3
      </div>
      <div>
        Alaleh Baghini
      </div>
      <div>
        39
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        4
      </div>
      <div>
        Jimit Patel
      </div>
      <div>
        35
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        5
      </div>
      <div>
        Norif Takaya
      </div>
      <div>
        42
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        6
      </div>
      <div>
        Carrie Grabowski
      </div>
      <div>
        30
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        7
      </div>
      <div>
        James Cameroon
      </div>
      <div>
        24
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        8
      </div>
      <div>
        Alaleh Baghini
      </div>
      <div>
        39
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        9
      </div>
      <div>
        Jimit Patel
      </div>
      <div>
        35
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        10
      </div>
      <div>
        Norif Takaya
      </div>
      <div>
        42
      </div>
    </li>
    <li
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div>
        11
      </div>
      <div>
        Carrie Grabowski
      </div>
      <div>
        30
      </div>
    </li>
  </ul>
</div>
`;
